lib/mail/utilities.rb

```diff
@@ -72,12 +72,25 @@ def dquote( str )
     #  unqoute(string) #=> 'This is "a string"'
     def unquote( str )
       if str =~ /^"(.*?)"$/
-        $1.gsub(/\\(.)/, '\1')
+        unescape($1)
       else
         str
       end
     end
 
+    # Removes any \-escaping.
+    #
+    # Example:
+    #
+    #  string = 'This is \"a string\"'
+    #  unescape(string) #=> 'This is "a string"'
+    #
+    #  string = '"This is \"a string\""'
+    #  unescape(string) #=> '"This is "a string""'
+    def unescape( str )
+      str.gsub(/\\(.)/, '\1')
```

Comment:
I was thinking something like this:

http://refiddle.com/refiddles/57ca78c875622d7947c32a00

The spec says that `\"` is the only escaping pair so all other `\` chars should be maintained.